app/hotels/src/HotelsContext.js

```diff
@@ -155,170 +156,57 @@ const validateInput = ({ checkin, checkout, cityId }: SearchInput) => {
   return validationError;
 };
 
-class Provider extends React.Component<Props, State> {
-  constructor(props: Props) {
-    super(props);
-
-    const checkin = getAsUtcDate(props.checkin);
-    const checkout = getAsUtcDate(props.checkout);
-    const cityId = props.cityId ?? null;
-    const errors = validateInput({ checkin, checkout, cityId });
-    this.state = {
-      hotelId: props.hotelId ?? '',
-      apiProvider: props.apiProvider,
-      version: props.version,
-      cityId,
-      checkin,
-      checkout,
-      roomsConfiguration: props.roomsConfiguration ?? null,
-      currency: props.currency,
-      cityName: props.cityName ?? null,
-      latitude: props.latitude ?? null,
-      longitude: props.longitude ?? null,
-      paymentLink: null,
-      currentSearchStats: defaultState.currentSearchStats,
-      getGuestCount: this.getGuestCount,
-      setHotelId: this.setHotelId,
-      setPaymentLink: this.setPaymentLink,
-      closeHotels: props.closeHotels,
-      actions: {
-        setCurrentSearchStats: this.setCurrentSearchStats,
-        setCheckinDate: this.setCheckinDate,
-        setCheckoutDate: this.setCheckoutDate,
-      },
-      errors: { ...errors },
-    };
-  }
-
-  setCurrentSearchStats = ({ priceMax, priceMin }: SetSearchStats) => {
-    this.setState({
-      currentSearchStats: {
-        priceMax: new Decimal(priceMax),
-        priceMin: new Decimal(priceMin),
-      },
-    });
+function Provider(props: Props) {
+  const cityId = props.cityId ?? null;
+
+  const [hotelId, setHotelId] = React.useState(props.hotelId ?? '');
+  const [paymentLink, setPaymentLink] = React.useState(null);
+  const [currentSearchStats, setCurrentSearchStats] = React.useState(
+    defaultState.currentSearchStats,
+  );
+
+  const dateFactoryState = useApi(dateFactory, {
```

Comment:
I don't know if `dateFactoryState` really helps me to understand better what's going on.
Perhaps, extracting `checkin, setCheckin, checkout, setCheckout` from `useApi(...)` would make it slightly more understandable that's it's a black box spitting out this information. 
Again, this is just a suggestion; I don't know what would be a better practice.

Reply:
> it's a black box spitting out this information

I see a function. Functions are either void or have a return type. We have the source code, and IMHO, it should be clear what is going on. 

I will however use destructuring here, since I did it on the line below anyway 🙂